enhancements/api-review/gcp_user_defined_labels.md

```diff
@@ -0,0 +1,610 @@
+  can configure a list of up to 32 user-defined labels in the OpenShift installer generated
+  install config, which is referenced and applied by installer and the in-cluster operators
+  on the GCP resources during creation.
+- Labels must be applied at creation time, in an atomic operation. Labels will not be added post cluster creation.
+- The administrator or service (in the case of Managed OpenShift) installing OpenShift can configure up to 50 tags
+  in the install config. The tags are added by installer and the in-cluster operators while creating GCP resources.
+
+### Non-Goals
+
+- Support for update or delete operation on labels or tags after creation of GCP resource.
+
+## Proposal
+
+***GCP Labels*** 
+
+A label of the form `kubernetes-io-cluster-<cluster_id>:owned` will be added to every
+resource created by OpenShift to enable administrator to differentiate the resources
+created for OpenShift cluster. An administrator is not allowed to add or modify the label
+having the prefix `kubernetes-io` or `openshift-io` in the key. The same can be found
+applied to other cloud platform resources which supports labels/tags for ex: AWS, Azure.
+
+New `userLabels` field will be added to `platform.gcp` of install-config for the user
+to define the labels to be added to the resources created by installer and in-cluster operators.
+
+If `platform.gcp.userLabels` in the install-config has any labels defined, then these labels
+will be added to all the GCP resources created by OpenShift, provided all the defined
+labels meet all the below conditions or else the cluster creation will fail.
+1. A label key and value must have minimum of 1 character and can have maximum of 63 characters.
+2. A label key and value must contain only lowercase letters, numeric characters,
+   underscores, and dashes.
+3. A label key must start with a lowercase letter.
+4. Each resource can have multiple labels, up to a maximum of 32.
+   - GCP supports a maximum of 64 labels per resource. Restricting the number of
+    user defined labels to 32 and reserving 32 for Openshift's internal use.
+
+All in-cluster operators that create GCP resources (Cluster Infrastructure ,Storage) will
+apply these labels during resource creation.
+
+The userLabels field is intended to be set at install time and is considered immutable.
+Components that respect this field must only ever add labels that they retrieve from this
+field to cloud resources, they must never remove labels from the existing underlying cloud
+resource even if the labels are removed from this field(despite it being immutable).
+
+If the userLabels field is changed post-install, there is no guarantee about how an
+in-cluster operator will respond to the change. Some operators may reconcile the
+change and update the labels on the GCP resource. Some operators may ignore the change.
+However, if labels are removed from userLabels, the label will not be removed from the
+GCP resource.
+
+***GCP Tags*** 
```

Comment:
One thing that's unclear to me, and perhaps you can clarify, if I have a tag, and I want to apply that to say a VM, can I apply multiple values for a tag to the VM, or only one value per tag?

Reply:
We can tag a VM with only one value for a key, it's scalar.

Reply:
Ok, please make sure to add a comment into the enhancement to explain that

Reply:
comment incorporated.